ONSdigital/eq-author-app
# EAR 1948 Option values for rad-chk-ans

What is the context of this PR?
- Add the capability to specify values for each option in a radio/checkbox/select answer types.

How to review
- Add to the list below as appropriate, including screenshots when necessary

Import the Capability examples questionnaire from pre-prod Author into your local environment and:
ensure it can be opened in Author;
then, ensure it can be viewed in Runner by pressing the view survey button
- Does this require a migration? We need one if existing JSON schema properties change
- Create a questionnaire with a radio, checkbox and select questions.
- Update settings to enable the Q Codes and Values tab
- Open to Q Codes and Values page and confirm layout is as per [design](https://www.figma.com/file/D0nxCipkp5HJxeF5FGl7fm/EAR---1948-Option-values-for-radio%2Fcheckbox%2Fselect-answers?type=design&node-id=0-1&mode=design&t=ZsCiolVJZfW8wUvG-0)
- Check that when values and q codes have been entered then validation messages are removed
- Check that when all validation messages have gone then the validation dot on the nav button also goes
- Check the survey export and convert schemas that option values are set
- Check that if an option value is missing then the label value is used instead

What to do after everything is green
- Bring this branch up-to-date with Origin/Master
- If there are a lot of commits or some are not helpful to read, squash them down
- Click the Merge button on this pull request
- Does this change mean the Capability examples questionnaire should be updated?
- Move the Jira ticket for this task into the next stage of the process
